src/plotting.py

```diff
@@ -1485,6 +1485,163 @@ def draw_fuzzy_beachball(problem, po):
         logger.info('Plot already exists! Please use --force to overwrite!')
 
 
+def draw_hudson(problem, po):
+    """
+    Modified from grond. Plot the hudson graph for the refrence (red) and
+    the best solution (biggest beachball). The position of the mean solution
+    is indicated by a black rectangle. Also a random number of models from the
+    selected stage are plotted as smaller beachballs on the hudson graph.
+    """
+    from pyrocko.plot import beachball, hudson
+    from pyrocko import moment_tensor as mtm
+    from numpy import random
+    if problem.config.problem_config.n_sources > 1:
+        raise NotImplementedError(
+            'Hudson plot is not yet implemented for more than one source!')
+
+    if not po.reference:
+        llk_str = po.post_llk
+        stage = Stage(homepath=problem.outfolder)
+
+        stage.load_results(
+            varnames=problem.varnames,
+            model=problem.model, stage_number=po.load_stage,
+            load='trace', chains=[-1])
+
+        n_mts = len(stage.mtrace)
+        m6s = num.empty((n_mts, 6), dtype='float64')
```

Comment:
U use this block 4 times, please make it a function point2array which takes the variable names as an argument to determine the column order of the resulting array

Reply:
Like this you mean?
def point2array(a, n_mts=None):

    if n_mts is not None:
        if str(a) is 'm6s':
            return num.empty((n_mts, 6), dtype='float64')
            
    if str(a) is 'm6s':
            return num.empty((1, 6), dtype='float64')

    if str(a) is 'm6_mean' or 'm6_max':
            return num.empty((6), dtype='float64')